client/allocrunner/taskrunner/script_check_hook.go

```diff
@@ -151,6 +151,22 @@ func (h *scriptCheckHook) upsertChecks() error {
 	return nil
 }
 
+// Exited runs when the task exits, but it may run again on restart.
+// Stop any running scripts but don't shutdown the script check since we may
+// still need it.
+func (h *scriptCheckHook) Exited(_ context.Context, _ *interfaces.TaskExitedRequest, _ *interfaces.TaskStopResponse) error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	// Cancel all running scripts, but don't close the shutdownCh since the
+	// task may still be restarted.
+	for _, script := range h.runningScripts {
+		script.cancel()
+	}
```

Comment:
This will cause script checks with `check.interval > restart.delay` to be run more frequently. I can't imagine why that would be a _bad thing_ since it just means we're heartbeating the TTL check more often, but I thought I'd mention it in case someone can dream up a way it could cause problems.